samza-core/src/main/java/org/apache/samza/operators/util/OperatorJsonUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.samza.operators.spec.OperatorSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OperatorJsonUtils {
```

Comment:
There's nothing JSON specific about this class. Also, an OperatorSpec should either provide access to its properties via getters, or return its string/json/map/object representation itself that other code can consume. Maybe just add `OperatorSpec#getSourceLocation()` to go along with `getOpCode` and `getOpId` for now?